app/code/RefreshMarkdownTask.php

```diff
@@ -78,40 +76,43 @@ private function getRepositories()
     }
 
     /**
-     * @param array $repository
+     * Clone $repository which contains the most current documentation source markdown files
      *
-     * @todo test this works with all modules
+     * @param array $repository
      */
     private function cloneRepository(array $repository)
     {
+
         list($remote, $folder, $branch) = $repository;
 
         $path = $this->getPath();
 
         exec("mkdir -p {$path}/src");
         exec("rm -rf {$path}/src/{$folder}_{$branch}");
-
-        $this->printLine("cloning " . $remote . "/" . $branch);
-
         chdir("{$path}/src");
-        exec("git clone -q https://github.com/{$remote}.git {$folder}_{$branch} --depth 1 --branch {$branch} --single-branch");
-
-        chdir("{$path}/src/{$folder}_{$branch}");
-    }
 
-    /**
-     * Clears out any non markdown files stored in assets
-     *
-     * @param array $repository
-     */
-    private function cleanRepository(array $repository)
-    {
-        $paths = array_merge(glob("*"), glob(".*"));
+        // If the dev=1 flag is used when RefreshMarkdownTask is run, a full git clone of the framework repository is kept
+        // to enable local development of framework and doc.silverstripe.org from within doc.silverstripe.org. Otherwise,
+        // only a shallow clone of the framework repository is made and all non-markdown files deleted, only allowing viewing
+        // of documentation files.
 
-        foreach ($paths as $path) {
-            if ($path !== "docs" && $path !== "." && $path !== "..") {
-                exec("rm -rf {$path}");
+        if( isset($_REQUEST['dev']) && $_REQUEST['dev'] ) {
```

Comment:
@SpiritLevel can you use the correct SilverStripe framework reference to these variables. i.e. $this->request->getVar('dev');